crates/ty_python_semantic/src/types.rs

```diff
@@ -1516,6 +1516,13 @@ impl<'db> Type<'db> {
                 true
             }
 
+            // Every `type[...]` is assignable to `type`
+            (Type::SubclassOf(_), Type::NominalInstance(instance))
+                if instance.class.is_known(db, KnownClass::Type) =>
+            {
+                true
+            }
```

Comment:
I doubt it matters much in this situation, but I'd prefer to express this in a slightly more general way:

```suggestion
            (Type::SubclassOf(_), _)
                if KnownClass::Type.to_instance(db).is_assignable_to(db, target) =>
            {
                true
            }
```

Reply:
> I doubt it matters much in this situation

Turns out it *does* matter. We have a few ecosystem results now. I think they look correct.